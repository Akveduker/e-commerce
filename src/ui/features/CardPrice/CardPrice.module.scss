@import "../../../styles/_mixins.scss";
.price {
    &__old {
        margin: 0;
        display: inline-block;
        font-size: 12px;
        @include fontSemibold;
        position: relative;
        color: $C1C;
        &::after {
            content: "";
            display: block;
            width: 100%;
            height: 1px;
            background-color: $C1C;
            position: absolute;
            top: 0;
            bottom: 0;
            margin: auto 0;
        }
    }
    &__green {
        color: $C2A;
    }
    &__big {
        .price__actual {
            font-size: 26px;
        }
    }
    &__actual {
        margin: 0;
        font-size: 18px;
        @include fontSemibold;
        line-height: 1.1;
    }
}
