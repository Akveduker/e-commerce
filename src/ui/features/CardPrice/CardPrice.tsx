import React, { FC } from 'react';
import { IProductDiscount } from '../../../types/Products/products';
import s from './CardPrice.module.scss'
interface CardPriceProps {
    isGreen?: boolean;
    isProductPage?: boolean;
    discount?: IProductDiscount | undefined;
    price: number;
}
const CardPrice: FC<CardPriceProps> = ({ discount, price, isProductPage, isGreen }) => {
    const discountPriceColor = isGreen ? `${s['price__actual']} ${s['price__green']}` : s['price__actual']
    const acutalPriceClass = isProductPage ? `${s['price']} ${s['price__big']}` : s['price']
    return (
        <div className={acutalPriceClass} data-testid={'container'}>
            {discount ?
                <div>
                    <p className={discountPriceColor} data-testid={'old'} >{discount.newPrice} USD</p>
                    <p className={s['price__old']} data-testid={'new'}>{price} USD</p>
                </div>
                :
                <div>
                    <p className={s['price__actual']}>{price} USD</p>
                </div>
            }
        </div>
    );
};

export default CardPrice;