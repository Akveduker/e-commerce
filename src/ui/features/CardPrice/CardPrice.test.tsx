import { render, screen } from "@testing-library/react"
import CardPrice from "./CardPrice"

describe('CardPrice', () => {
    describe('Со скидкой', () => {
        const element = <CardPrice discount={{ newPrice: 32, percent: 20 }} price={40} />
        test('Снапшот', () => {
            const { container } = render(element)
            expect(container).toMatchSnapshot()
        })
        test('Есть блок новой цены', () => {
            render(element)
            expect(screen.getByTestId('new')).toBeInTheDocument()
        })
    })
    describe('без скидки', () => {
        const element = <CardPrice price={40} />
        test('Снапшот', () => {
            const { container } = render(element)
            expect(container).toMatchSnapshot()
        })
        test('Нет блока новой цены', () => {
            render(element)
            expect(screen.queryByTestId('new')).toBeNull()
        })
    })
    describe('Модификация класса', () => {
        test('Класса нет', () => {
            render(<CardPrice price={40} discount={{ newPrice: 32, percent: 20 }} />)
            const container = screen.queryByTestId('container')
            const oldPrice = screen.queryByTestId('old')
            expect(container).not.toHaveClass('price__green')
            expect(oldPrice).not.toHaveClass('price__big')
        })
        test('Класс есть', () => {
            render(<CardPrice isGreen isProductPage price={40} discount={{ newPrice: 32, percent: 20 }} />)
            const container = screen.getByTestId('container')
            const oldPrice = screen.getByTestId('old')
            expect(container).toHaveClass('price__big')
            expect(oldPrice).toHaveClass('price__green')
        })
    })
})