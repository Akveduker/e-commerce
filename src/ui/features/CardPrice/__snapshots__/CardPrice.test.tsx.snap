// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardPrice Со скидкой Снапшот 1`] = `
<div>
  <div
    class="price"
    data-testid="container"
  >
    <div>
      <p
        class="price__actual"
        data-testid="old"
      >
        32
         USD
      </p>
      <p
        class="price__old"
        data-testid="new"
      >
        40
         USD
      </p>
    </div>
  </div>
</div>
`;

exports[`CardPrice без скидки Снапшот 1`] = `
<div>
  <div
    class="price"
    data-testid="container"
  >
    <div>
      <p
        class="price__actual"
      >
        40
         USD
      </p>
    </div>
  </div>
</div>
`;
